mutiny-core/src/nostr/mod.rs

```diff
@@ -98,14 +98,27 @@ impl<S: MutinyStorage> NostrManager<S> {
         relays
     }
 
-    pub fn get_nwc_filters(&self) -> Vec<Filter> {
-        self.nwc
+    pub fn get_nwc_filters(&self) -> Result<Vec<Filter>, MutinyError> {
+        // if we haven't synced before, use now and save to storage
+        let time_stamp = match self.storage.get_nwc_sync_time()? {
```

Comment:
Wasn't dm sync time also added? Wouldn't that be the same there too?

Reply:
That's in the other pr, not sure we'd want to have them use the same timestamp. We could process a dm but not a nwc and miss it.

Reply:
But NWCs are DM's? If we treat it all the same as sync all the DM's before marking the sync time, wouldn't that be the same?

Reply:
The worry is that we sync the DMs but not sync NWCs to completion and then set the sync time? So perhaps we wait for them all to sync before marking the DMs? Afterall, syncing the DMs is so we can get all of the payment requests anyways, there's functionally no difference in why we are syncing them.

Reply:
It's more so because they exist on different relays and filters so the ordering isn't guaranteed to be the same for both. The Damus relays could be giving us dms from a week ago but we get an nwc from the mutiny relays from today, we'll process the nwc and have the potential to miss all the dms if we close before we finish processing

Reply:
This is a symptom of a bigger problem then and it's syncing events across relays across nostr keys. Your same argument applies to getting events from one NWC and processing that NWC but it still has other NWCs to go through.

Reply:
Yeah the way we do it on blastr is best but we don't be able to do that with real relays

Reply:
I guess we need something, can try it out and see how it goes.